File: src/main/java/sprykerFileCreator/model/definition/spryker/ClientDependencyProviderDefinition.java
package sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class ClientDependencyProviderDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.CLIENT_DEPENDENCY_PROVIDER;
    }

    @Override
    public String getNamePattern() {
        return "{{ bundleName }}DependencyProvider";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Client\\{{ bundleName }}";
    }

    @Override
    public String getMethodForReturnType() {
        return "";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {};

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Client\\Kernel\\AbstractBundleDependencyProvider";
    }
}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/definitions/spryker/ClientDependencyProviderDefinition.kt
package com.github.tobi812.sprykerplugin.models.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface

class ClientDependencyProviderDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.CLIENT_DEPENDENCY_PROVIDER
    override val namePattern: String
        get() = "{{ bundleName }}DependencyProvider"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Client\\{{ bundleName }}"
    override val methodForReturnType: String
        get() = ""
    override val docBlockClasses: Array<String>
        get() = arrayOf()
    override val defaultParentClass: String
        get() = "Spryker\\Client\\Kernel\\AbstractBundleDependencyProvider"
}